src/core/bitmap_type.hpp

```diff
@@ -0,0 +1,119 @@
+/* $Id$ */
+
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file bitmap_type.hpp Bitmap functions. */
+
+#ifndef BITMAP_TYPE_HPP
+#define BITMAP_TYPE_HPP
+
+#include "smallvec_type.hpp"
+
+class Bitmap : public SmallVector<uint8, 8> {
```

Comment:
Why this over std::bitset? Also almost entirely missing doxymentation (which should include why it exists over std::bitset)

Reply:
The size of a std::bitset<> is set at compile time.

Reply:
Actually it doesn't need to use SmallVector at all, because I didn't end up using it that way.